### ISSUE TITLE
Unable to download the latest version

### ISSUE THREAD

**Issue by theAguy** [OpenGene/fastp#383]
Dear all,

I am using Mac M1 processor and it is giving me a hard time updating FASTP without conda.
With conda I am able to update the FASTP however the latest version is 0.19

It will be admirable if you will be able to publish the latest version in conda as well.

Best regards and thank you.

**Comment by sfchen** [comment]
The latest version is already in Conda.

Each new fastp version is in Conda.

**Comment by theAguy** [comment]
Thank you for your email.

However, when running conda install -c bioconda fastp
It's installing version 0.12.4-0

And while running conda install -c bioconda/label/cf201901 fastp
It's installing version 0.19.5
As you can see:
[image: image.png]

Both are not the latest version, and lacking some essential properties for
my work.

I would highly appreciate your help!

Thank you



‫בתאריך יום א׳, 12 בדצמ׳ 2021 ב-13:01 מאת ‪Shifu Chen‬‏ <‪
***@***.***‬‏>:‬

**Comment by sfchen** [comment]
See https://anaconda.org/bioconda/fastp

**Comment by angusdavison** [comment]
I have got exactly the same problem, the link above (despite what it says) does not install 0.23.2

**Comment by jakobnissen** [comment]
@angusdavison Are you sure you're not seeing a dependency conflict? Fastq requires a newer version of libdeflate, which means that if you have e.g. BWA in your environment, you can't get the latest version of fastp.

Can you install it in an empty environment with nothing else it in?

**Comment by angusdavison** [comment]
I dont think that is the issue, though getting somewhat beyond my skills/knowledge. In any case, I installed the binaries via "wget http://opengene.org/fastp/fastp" - that is working fine.  Thanks for your consideration

**Comment by sfchen** [comment]
> @angusdavison Are you sure you're not seeing a dependency conflict? Fastq requires a newer version of libdeflate, which means that if you have e.g. BWA in your environment, you can't get the latest version of fastp.
> 
> Can you install it in an empty environment with nothing else it in?

What libdeflate version does BWA use?

**Comment by jakobnissen** [comment]
Looking at it for a bit more, I _think_ the conflict is actually samtools not bwa, which needs htslib, which uses libdeflate 1.6, whereas fastp uses 1.7. But it's hard to know exactly.

**Comment by angusdavison** [comment]
Yes, agree there is probably a conflict but I don't know the source - a force install of 0.23.2 produced the following errors. Not sure if this is helpful or not (and I have got fastp working so I am not concerned)

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package libgcc-ng conflicts for:
python=3.9 -> libgcc-ng[version='>=7.3.0|>=7.5.0']
python=3.9 -> zlib[version='>=1.2.11,<1.3.0a0'] -> libgcc-ng[version='>=7.2.0']

Package libstdcxx-ng conflicts for:
python=3.9 -> libstdcxx-ng[version='>=7.3.0|>=7.5.0']
fastp=0.23.2 -> libstdcxx-ng[version='>=9.4.0']The following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.17=0
  - python=3.9 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']

Your installed version is: 2.17

**Comment by sfchen** [comment]
But fastp dons't declare a dependency on libstdcxx-ng[version='>=9.4.0']. 

I checked annocanda, the latest version for libstdcxx-ng is 9.3.0.

**Comment by Jediworm** [comment]
Hello. Unfortunately, I am also having some install issues.

"conda install -c bioconda fastp" installs version 0.12.4

"conda install -c bioconda/label/cf201901 fastp" installs version 0.19.5

I would like to use these two arguments: --detect_adapter_for_pe and --cut_tail. However, it looks like --detect_adapter_for_pe isn't in 0.12.4 and --cut_tail isn't in 0.19.5. As an example, I get the message "undefined option: --cut_tail"

I created a new conda environment to ensure there were no conflicts and the problem is still there. I am running conda 4.11.0 on OSX through terminal.

Any guidance would be appreciated. Thanks!

**Comment by Jediworm** [comment]
Hello again. Just wanted to provide an update for those looking for a workaround. I was able to install an older version of fastp by downloading the file from conda and running: "conda install fastp-0.20.0-hd9629dc_0.tar.bz2". I was then able to use the two arguments I wanted. Importantly, it wouldn't work on the university cluster for some reason (cannot execute binary file) but it worked on my work macbook. I tried it the same thing with the latest version 0.23.2 but it didn't work (dyld: Library not loaded: @rpath/libisal.2.dylib). Just trialling a few more: version 0.23.0 also didn't work with this method but 0.22.0 did.

Thanks!

**Comment by crcardenas** [comment]
I am also having an issue with fastp where it wont write cleaned reads. Using the basic commands with paired end data, fastp only writes QC reports. I know there are good reads for all the sequences when reading the the json & html outputs; but this is a seperate issue. To solve this, I thought I would try updating it to the latest version since I realized I did not have the latest. However, I cannot install the latest 0.23.# or even 0.22.# versions

_conda install -c bioconda/label/cf201901_ installs fastp-**0.19.5**
_conda install -c bioconda/label/main_ installs fastp-**0.20.1**
_conda install -c bioconda fastp_ installs fastp-**0.20.1**

I've tried various conda install commands to force install the absolute latest version "fastp=0.23.2". I've used a fresh conda environment with different python versions. At this point I'm not sure what to do.

@Jediworm 

I wonder if this workaround is OK if you have sudo permissions. I've tried downloading it to my home dir on the cluster I use and installing from there.
```
$ fastp -v
fastp: error while loading shared libraries: libisal.so.2: cannot open 
shared object file: No such file or directory
```
From my novice understanding, don't I need to download the tarball to the anaconda package directory? (/local/anaconda3/pkgs/); and then "conda install /local/anaconda3/pkgs/fastp-...." ?

**Comment by lcscs12345** [comment]
A workaround would be creating a new conda environment and installing the latest fastp.
`conda create -n fastp -c bioconda fastp=0.23.2`

**Comment by mailmrcai** [comment]
I have the same issue, solved by setting the environment variable.
export LD_LIBRARY_PATH=" your libdeflate path containing the file libdeflate.so.0"
I wonder this is caused by makefile
